operators/src/source/gdal_source.rs

```diff
@@ -372,6 +374,155 @@ impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
     }
 }
 
+#[derive(PartialEq, Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct GdalMetadataFixedTimes {
+    pub time_steps: Vec<TimeInterval>,
+    pub params: GdalDatasetParameters,
+    pub result_descriptor: RasterResultDescriptor,
+    pub time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+}
+
+impl GdalMetadataFixedTimes {
+    #[allow(dead_code)]
+    fn new(
+        time_steps: Vec<TimeInterval>,
+        params: GdalDatasetParameters,
+        result_descriptor: RasterResultDescriptor,
+        time_placeholders: HashMap<String, GdalSourceTimePlaceholder>,
+    ) -> Self {
+        let mut sorted_time_steps = time_steps;
+        sorted_time_steps.sort_by(|a, b| a.end().partial_cmp(&b.start()).unwrap());
+
+        GdalMetadataFixedTimes {
+            time_steps: sorted_time_steps,
+            params,
+            result_descriptor,
+            time_placeholders,
+        }
+    }
+}
+
+#[async_trait]
+impl MetaData<GdalLoadingInfo, RasterResultDescriptor, RasterQueryRectangle>
+    for GdalMetadataFixedTimes
+{
+    async fn loading_info(&self, query: RasterQueryRectangle) -> Result<GdalLoadingInfo> {
+        let minimum_step = TimeStep {
+            granularity: TimeGranularity::Millis,
+            step: 1,
+        };
+
+        let sorted_time_steps = if self
+            .time_steps
+            .windows(2)
+            .all(|w| w[0].end() <= w[1].start())
+        {
+            self.time_steps.clone()
+        } else {
+            let mut sorted_time_steps = self.time_steps.clone();
+            sorted_time_steps.sort_by(|a, b| a.end().partial_cmp(&b.start()).unwrap());
+            sorted_time_steps
+        };
+
+        let mut time_steps_no_gaps = sorted_time_steps.clone();
```

Comment:
you should fill the gaps in the `new` method instead of every time `loading_info` is called.

also: how do you distinguish between gaps and data? In a gap, we should directly create NO-DATA-tiles instead of requesting the server with the a time where we know there is no data.

Reply:
I added a new field in `GdalMetadataFixedTimes` to store the time steps with no gaps and made all fields private, so now everything can be done in the `new` method.